staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package buffered implements the audit.Backend interface with buffered channel.
+package buffered
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+	auditinternal "k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/audit"
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+const (
+	// Default configuration values for ModeBatch.
```

Comment:
Some comments in this file should be updated.

Reply:
+1, please go through all comments in this file

Reply:
Yes, I see. But it is indeed modeBatch now.

Reply:
bufferedBackend may be more appropriate